cmd/sync/sync.go

```diff
@@ -295,6 +309,102 @@ func main() {
 				continue
 			}
 
+			// Check if the header is needed
+			copyHeader := config.CheckHeader()
+			//nolint:nestif
+			if copyHeader {
+				// Get the header from db
+				header, err := db.GetHeaderStableId(message.AccessionID)
+				if err != nil {
+					log.Errorf("GetHeaderStableId failed "+
+						"(corr-id: %s, "+
+						"filepath: %s, "+
+						"user: %s, "+
+						"accessionid: %s, "+
+						"decryptedChecksums: %v, error: %v)",
+						delivered.CorrelationId,
+						message.Filepath,
+						message.User,
+						message.AccessionID,
+						message.DecryptedChecksums,
+						err)
+				}
+
+				// Decrypt header
+				log.Debug("Decrypt header")
+				DecrHeader, err := FormatHexHeader(header, *key)
+				if err != nil {
+					log.Errorf("Failed to decrypt the header %s "+
+						"(corr-id: %s, "+
+						"filepath: %s, "+
+						"user: %s, "+
+						"accessionid: %s, "+
+						"decryptedChecksums: %v, error: %v)",
+						filePath,
+						delivered.CorrelationId,
+						message.Filepath,
+						message.User,
+						message.AccessionID,
+						message.DecryptedChecksums,
+						err)
+
+					if e := delivered.Nack(false, true); e != nil {
+						log.Errorf("Failed to NAck because of decrypt header failed "+
+							"(corr-id: %s, "+
+							"filepath: %s, "+
+							"user: %s, "+
+							"accessionid: %s, "+
+							"decryptedChecksums: %v, error: %v)",
+							delivered.CorrelationId,
+							message.Filepath,
+							message.User,
+							message.AccessionID,
+							message.DecryptedChecksums,
+							e)
+					}
+				}
+
+				// Reencrypt header
+				log.Debug("Reencrypt header")
+				newHeader, err := reencryptHeader(*key, *publicKey, *DecrHeader)
+				if err != nil {
+					log.Errorf("Failed to reencrypt the header %s "+
+						"(corr-id: %s, "+
+						"filepath: %s, "+
+						"user: %s, "+
+						"accessionid: %s, "+
+						"decryptedChecksums: %v, error: %v)",
+						filePath,
+						delivered.CorrelationId,
+						message.Filepath,
+						message.User,
+						message.AccessionID,
+						message.DecryptedChecksums,
+						err)
+
+					if e := delivered.Nack(false, true); e != nil {
+						log.Errorf("Failed to NAck because of reencrypt header failed "+
+							"(corr-id: %s, "+
+							"filepath: %s, "+
+							"user: %s, "+
+							"accessionid: %s, "+
+							"decryptedChecksums: %v, error: %v)",
+							delivered.CorrelationId,
+							message.Filepath,
+							message.User,
+							message.AccessionID,
+							message.DecryptedChecksums,
+							e)
+					}
+				}
+
+				// write header to destination file
+				_, err = dest.Write(newHeader)
+				if err != nil {
+					log.Fatalf("Error in header writer: %v", err)
+				}
```

Comment:
This will kill the app when triggered.
Instead a message should be sent like you do above

Reply:
Right, I have missed it